pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -401,12 +401,19 @@ def save_checkpoint(self, filepath, weights_only: bool = False):
             if self.trainer.training_type_plugin:
                 checkpoint = self.trainer.training_type_plugin.on_save(checkpoint)
             try:
-                atomic_save(checkpoint, filepath)
+                atomic_save(checkpoint, filepath, **self._storage_options)
             except AttributeError as err:
                 if LightningModule.CHECKPOINT_HYPER_PARAMS_KEY in checkpoint:
                     del checkpoint[LightningModule.CHECKPOINT_HYPER_PARAMS_KEY]
                 rank_zero_warn(
                     'Warning, `hyper_parameters` dropped from checkpoint.'
                     f' An attribute is not picklable {err}'
                 )
-                atomic_save(checkpoint, filepath)
+                atomic_save(checkpoint, filepath, **self._storage_options)
+
+    @property
+    def _storage_options(self):
+        try:
+            return self.trainer.checkpoint_callback.storage_options
+        except AttributeError:
+            return {}
```

Comment:
this will not work if we have multiple checkpoint callbacks with different storage options. imagine a setup where you periodically save checkpoints as `last.ckpt` during training with a short TTL, but you also have another checkpoint callback to save the best K models at the end of validation